### PR TITLE
don't merge: feat: allow drawbridge sourced toml files to be editable

### LINKED ISSUE #91: Allow the Enarx.toml be to be editable when deploying from drawbridge
blocked until #68 is merged

`enarx deploy` doesn't allow us to change the Enarx.toml configuration to change ports/arguments. We should instead try to download the main.wasm/Enarx.toml and use `enarx run` instead.

This will simplify things and allow users to avoid port conflict errors and change arguments on a per workload basis.

We should also handle refreshing the Enarx.toml on the frontend via a button and disallow deployments if the Enarx.toml can not be pulled.

### PR DESCRIPTION
**Pull request by definitelynobody**
Closes #91

If the toml file was not changed or couldn't be read then 'enarx deploy' will be used instead.

The only change the user will see with this change is that the config is always editable.

### CODE REVIEW

File: src/jobs.rs [hunk 1]
```diff
@@ -70,11 +104,11 @@ impl Job {
         COUNT.load(Ordering::SeqCst)
     }
 
-    pub fn new(
+    pub async fn new<'a>(
```
Comment (rvolosatovs) [inline src/jobs.rs:107]: Is it really `async`? Is the lifetime used anywhere?

File: src/jobs.rs [hunk 2]
```diff
@@ -87,18 +121,29 @@ impl Job {
...
+
+                match &toml {
+                    None => enarx_deploy(&cmd, slug)?,
+                    Some(toml) => {
+                        let slug = Slug::new(slug.clone())
+                            .ok_or_else(|| StatusCode::BAD_REQUEST.into_response())?;
+                        let wasm_bytes = slug
+                            .read("main.wasm")
+                            .await
+                            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR.into_response())?
```
Comment (rvolosatovs) [inline src/jobs.rs:133]: These errors will be impossible to debug in production settings. How about we don't discard the error, but rather log it?
Something like this:
```suggestion
                            .map_err(|e| {
                            	warn!("failed to fetch Wasm: {e}"); // or debug!, error!, whichever is appropriate
                            	StatusCode::INTERNAL_SERVER_ERROR.into_response()
                            })?
```

File: src/jobs.rs [hunk 3]
```diff
@@ -87,18 +121,29 @@ impl Job {
...
+                            .bytes()
+                            .await
+                            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR.into_response())?;
```
Comment (rvolosatovs) [inline src/jobs.rs:136]: Are these errors really Internal Server Errors? Isn't the slug user input?

Internal Server Errors are only for when the Benefice server itself has an issue (e.g. a bug or it instance ran out of space and could not create a temporary file)

File: src/jobs.rs [hunk 4]
```diff
@@ -87,18 +121,29 @@ impl Job {
...
+                        let mut new_wasm = tempfile::NamedTempFile::new()
+                            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR.into_response())?;
+                        new_wasm
+                            .write_all(&wasm_bytes)
+                            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR.into_response())?;
+                        let child = enarx_run(&cmd, toml, &new_wasm)?;
+                        wasm = Some(new_wasm);
```
Comment (rvolosatovs) [inline src/jobs.rs:143]: Instead of mutating `wasm`, return it from the `match`

File: src/main.rs [hunk 5]
```diff
@@ -548,30 +548,11 @@ async fn root_post(
...
-) -> Result<reqwest::Response, reqwest::Error> {
-    HTTP.get(&format!(
-        "https://store.profian.com/api/v0.2.0/{repo}/_tag/{tag}/tree/Enarx.toml"
-    ))
-    .send()
-    .await
-}
-
-async fn enarx_toml_fallback(
+async fn enarx_toml_fallback<'a>(
```
Comment (rvolosatovs) [inline src/main.rs:551]: When is this called? Can we add some docs?

What is the `'a` lifetime for?

**Review by rvolosatovs — CHANGES_REQUESTED** [review]

**Comment by definitelynobody** [comment]
I will make these changes and apply them in #118

**Pull request closed without merging**